torchtune/models/gemma2/_attention.py

```diff
@@ -0,0 +1,626 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+from typing import Optional
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+from torchtune.modules.attention_utils import _MaskType
+from torchtune.modules.kv_cache import KVCache
+
+# from torchtune.utils._import_guard import _SUPPORTS_FLEX_ATTENTION
+# The flex attention implementation for gemma2 is not working yet
+# flex attention is disabled for now untill we solve the case
+_SUPPORTS_FLEX_ATTENTION = False
+
+if _SUPPORTS_FLEX_ATTENTION:
+    from torch.nn.attention.flex_attention import create_block_mask
+    from torchtune.models.gemma2._attention_utils import (
+        compile_friendly_flex_attention_with_score_and_block,
+        flex_causal_sliding_window,
+        flex_tanh_soft_capping_with_scaling,
+    )
+logger = logging.getLogger(__name__)
```

Comment:
nit: Why is this style of logger getting proliferated? We should be calling `get_logger` from our utils.

Reply:
It's just a copy paste on my side, let me know if you want me to change that on this PR.

Reply:
You can just change to `torchtune.utils.get_logger`, but no strong preference here. Either way we should clean up other usages in a follow-up